### PR TITLE
Feature/condo/sberdoma 326/employee profile view

### PR DESCRIPTION
**Pull request by Dimitreee**

### CODE REVIEW

**Comment by Dimitreee** [comment]
hide sidenav while employee is blocked

**Comment by AntonAL** [comment]
Please, make user avatar clickable on top-right corner, — it should follow to user profile page, as user label does.

**Comment by AntonAL** [comment]
Please, add a redirect to a user profile view page, when a form is successfully submitted on a user profile edit page.

**Review by dkoviazin — APPROVED** [review]
will need to check that we use the same access condition for blocked users after merging this pr to master

File: apps/condo/domains/common/components/CountDownTimer.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        }
+    }
+}
+
+type CountDownChildrenType = (
+    // TODO(Dimitreee): remove any
+    { countdown, runAction, loading }: { countdown: number, runAction: () => Promise<any>, loading: boolean }
+) => JSX.Element
+
+interface ICountdownTimer {
```
Comment (pahaz) [inline apps/condo/domains/common/components/CountDownTimer.tsx:51]: ICountDownTimer

File: apps/condo/domains/common/components/CountDownTimer.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    // TODO(Dimitreee): remove any
+    action: (...args: any[]) => Promise<any>
+    id: string
+    timeout?: number
+    children: CountDownChildrenType
+}
+
+export const CountdownTimer: React.FC<ICountdownTimer> = (props) => {
```
Comment (pahaz) [inline apps/condo/domains/common/components/CountDownTimer.tsx:59]: CountDownTimer

File: apps/condo/domains/common/components/containers/BaseLayout/components/SideMenu.tsx [hunk 3]
```diff
@@ -97,6 +99,12 @@ const TicketCreateButton = () => {
 
 export const SideMenu: React.FC<ISideMenuProps> = (props) => {
     const { onLogoClick, menuData, isMobile, isSideMenuCollapsed, toggleSideMenuCollapsed } = props
+    const organization = useOrganization()
+    const isEmployeeBlocked = get(organization, ['link', 'isBlocked'], false)
+
+    if (isEmployeeBlocked) {
```
Comment (pahaz) [inline apps/condo/domains/common/components/containers/BaseLayout/components/SideMenu.tsx:105]: Why do you want to hide the side menu for blocked users? probably it's better to patch OrganizationRequired to check it and redirect the user to some page.

Comment (pahaz) [reply apps/condo/domains/common/components/containers/BaseLayout/components/SideMenu.tsx:105]: This behavior is confusing the user.

File: apps/condo/domains/common/components/containers/FormList.tsx [hunk 4]
```diff
@@ -398,7 +402,9 @@ const FormWithAction: FunctionComponent<IFormWithAction> = (props) => {
     }, [])
 
     function handleSave () {
-        form.submit()
+        if (form.isFieldsTouched()) {
```
Comment (pahaz) [inline apps/condo/domains/common/components/containers/FormList.tsx:405]: 🥇

File: apps/condo/domains/common/components/containers/FormTableBlocks.tsx [hunk 5]
```diff
@@ -25,8 +27,8 @@ const _USE_TABLE_INITIAL_STATE = {
 
 function createNewGQLItem () {
     return {
-        id: Math.random(),
```
Comment (pahaz) [inline apps/condo/domains/common/components/containers/FormTableBlocks.tsx:28]: Can this cause problems in runtime?

File: apps/condo/domains/organization/access/OrganizationEmployeeRole.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ async function canReadOrganizationEmployeeRoles ({ authentication: { item: user
     if (user.isAdmin) return {}
     return {
         // user is inside employee list
-        organization: { employees_some: { user: { id: user.id } } },
+        organization: { employees_some: { user: { id: user.id }, isBlocked: false } },
```
Comment (pahaz) [inline apps/condo/domains/organization/access/OrganizationEmployeeRole.js:11]: I think that you can read your Employee, but you can't change it

File: apps/condo/domains/organization/access/OrganizationEmployeeRole.js [hunk 7]
```diff
@@ -25,7 +25,13 @@ async function canManageOrganizationEmployeeRoles ({ authentication: { item: use
         const employeeForUser = await getByCondition('OrganizationEmployee', {
             organization: { id: originalInput.organization.connect.id },
             user: { id: user.id },
+            deletedAt: null,
```
Comment (pahaz) [inline apps/condo/domains/organization/access/OrganizationEmployeeRole.js:28]: 🥇

File: apps/condo/domains/organization/components/EmployeeInviteRetryButton.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+interface IEmployeeInviteRetryButtonProps {
+    employee: OrganizationEmployee
+}
+
+export const EmployeeInviteRetryButton: React.FC<IEmployeeInviteRetryButtonProps> = (props) => {
+    const intl = useIntl()
+    const RetryInviteMessage = intl.formatMessage({ id: 'employee.RetryInvite' })
+    const NotImplementedYetMessage = intl.formatMessage({ id: 'NotImplementedYet' })
+    const Seconds = intl.formatMessage({ id: 'Seconds' })
+    const { link } = useOrganization()
```
Comment (pahaz) [inline apps/condo/domains/organization/components/EmployeeInviteRetryButton.tsx:19]:
```suggestion

    const { link } = useOrganization()
```

File: apps/condo/domains/organization/components/EmployeeInviteRetryButton.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+    const { employee } = props
+    const isEmployeeReinvitable = canReinviteEmployee(link, employee)
+
+    const resetEmployeeInvite = () => {
+        if (!isEmployeeReinvitable) {
+            return Promise.resolve()
+        }
+
+        console.log('resetEmployeeInvite')
```
Comment (pahaz) [inline apps/condo/domains/organization/components/EmployeeInviteRetryButton.tsx:29]: log

File: apps/condo/domains/organization/components/EmployeeProfileForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+export const EmployeeProfileForm = () => {
+    const intl = useIntl()
+    const { query } = useRouter()
+
+    const ApplyChangesMessage = intl.formatMessage({ id: 'ApplyChanges' })
+    const ProfileUpdateTitle = intl.formatMessage({ id: 'profile.Update' })
+    const RoleLabel = intl.formatMessage({ id: 'employee.Role' })
+    const PositionLabel = intl.formatMessage({ id: 'employee.Position' })
+
```
Comment (pahaz) [inline apps/condo/domains/organization/components/EmployeeProfileForm/index.tsx:35]:
```suggestion
    const intl = useIntl()
    const ApplyChangesMessage = intl.formatMessage({ id: 'ApplyChanges' })
    const ProfileUpdateTitle = intl.formatMessage({ id: 'profile.Update' })
    const RoleLabel = intl.formatMessage({ id: 'employee.Role' })
    const PositionLabel = intl.formatMessage({ id: 'employee.Position' })

    const { query } = useRouter()

```

File: apps/condo/domains/organization/utils/serverSchema/guards.js [hunk 11]
```diff
@@ -6,6 +6,7 @@ const checkEmployeeExistency = async (context, organization, email, phone, user)
     const employeesByEmail = await findOrganizationEmployee(context, {
         email,
         organization: { id: organization.id },
+        deletedAt: null,
```
Comment (pahaz) [inline apps/condo/domains/organization/utils/serverSchema/guards.js:9]: 🥇

File: apps/condo/domains/common/components/containers/FormTableBlocks.tsx (hunk 5 above)
Comment (Dimitreee) [reply apps/condo/domains/common/components/containers/FormTableBlocks.tsx:28]: There were no errors seen by me

**Comment by AntonAL** [comment]
When I opening `/user/update` and just click form submit button, nothing happens.
I expect some feedback from the system.
When no changes are made in form fields, I propose one of the following:
1. Redirect to user profile, as it does after successful form submission;
2. Block submit button.

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/organization/components/EmployeeProfileForm/index.tsx [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    const { query } = useRouter()
+
+    const { obj: employee, loading, error, refetch } = OrganizationEmployee.useObject({ where: { id: String(get(query, 'id', '')) } })
+    const updateEmployeeAction = OrganizationEmployee.useUpdate({}, () => refetch())
+
+    if (error) {
+        return <LoadingOrErrorPage title={'Title'} loading={loading} error={error ? 'Error' : null}/>
```
Comment (pahaz) [inline apps/condo/domains/organization/components/EmployeeProfileForm/index.tsx:40]: ?

File: apps/condo/domains/organization/utils/clientSchema/OrganizationEmployee.ts [hunk 13]
```diff
@@ -41,10 +42,37 @@ function convertToUIFormState (state: IOrganizationEmployeeUIState): IOrganizati
...
+
+    // TODO(Dimitreee): refactor client utils to add disconectFeature, think about solution
+    for (const fieldName of Object.keys(state)) {
+        if (state[fieldName] && RELATIONS.includes(fieldName)) {
+            const disconnectId = get(state, [fieldName, 'disconnectId'])
+            const id = get(state, [fieldName, 'id'])
+
+            if (disconnectId) {
+                result[fieldName] = {
+                    disconnect: {
```
Comment (pahaz) [inline apps/condo/domains/organization/utils/clientSchema/OrganizationEmployee.ts:54]: disconnectAll: true

**Pull request merged**